### PR TITLE
More fine-grained entry points in deposit step

### LINKED ISSUE #159: More fine-grained entry points for deposit step
Currently, the deposit step only has one entry point: `hermes.deposit`. It should be split into multiple entry points to give users more opportunities to customize hermes' behavior and make it easier for them to reuse existing code. Potential entry points are:

- metadata deposition (create a new deposit)
- metadata update (an existing set of metadata is updated) (is there a difference between this one and the initial deposition?)
- deletion of existing artifacts
- file/artifact deposition (upload of files to the deposit)
- final publish step (like clicking Publish on the website)

"Grouping" of these steps should also be considered. E.g.

- hermes.publication.prepare_deposit
- hermes.publication.map_metadata
- hermes.publication.deposition.add_metadata?
- hermes.publication.deposition.update_metadata
- hermes.publication.deposition.delete_artifacts
- hermes.publication.deposition.upload_artifacts
- hermes.publication.deposition.publish

### PR DESCRIPTION
**Pull request by zyzzyxdonta**
This merge request implements more fine-grained entry points for users of hermes to hook into. The "namespacing" of the entry point groups is ~~modeled after [this](https://user-images.githubusercontent.com/3007126/207607703-d34156b9-be70-468d-8898-269c5b90b132.png) plan.~~ not final and might have to be re-thought for v1.0.

Since entry points were changed in this merge request, `poetry install` has to be run to test it.

Closes #159

### CODE REVIEW

File: src/hermes/commands/workflow/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+# SPDX-FileContributor: David Pape
+
+from .clean import clean
+from .curate import curate
+from .deposit import deposit
+from .harvest import harvest
+from .postprocess import postprocess
+from .process import process
```
Comment (led02) [inline src/hermes/commands/workflow/__init__.py:12]: This seems to confuse Sphinx AutoDoc. Is this propagation required?

**Review by led02 — COMMENTED** [review]
Please see inline comments... Review is not completed yet

File: src/hermes/commands/workflow/__init__.py (hunk 1 above)
Comment (zyzzyxdonta) [reply src/hermes/commands/workflow/__init__.py:12]: It is not required. But if we do it like this, we can keep [this block as is](https://github.com/hermes-hmc/workflow/blob/e5ad51d1498f3a82266a8e4b9b35254572506d88/src/hermes/cli.py#L165). The alternative would be to have

```python
main.add_command(workflow.clean.clean)
main.add_command(workflow.harvest.harvest)
main.add_command(workflow.process.process)
main.add_command(workflow.curate.curate)
main.add_command(workflow.deposit.deposit)
main.add_command(workflow.postprocess.postprocess)
```

in the CLI module.

File: .flake8 [hunk 2]
```diff
@@ -5,3 +5,5 @@
 [flake8]
 max-complexity = 15
 max-line-length = 120
+per-file-ignores =
+    */__init__.py: F401
```
[the remark this answers is not in the document]
Comment (zyzzyxdonta) [reply .flake8]: Done in 1197ac3ce8c4006c392378de51ec66d56f0e05c2

**Comment by zyzzyxdonta** [comment]
I'm going to rebase later

File: src/hermes/commands/workflow/__init__.py (hunk 1 above)
Comment (zyzzyxdonta) [reply src/hermes/commands/workflow/__init__.py]: I decided not to split the code into multiple files in this PR. So this was removed again.

File: pyproject.toml [hunk 3]
```diff
@@ -86,14 +86,29 @@ git = "hermes.commands.process.git:process"
...
+invenio = "hermes.commands.deposit.invenio:update_metadata"
+
+[tool.poetry.plugins."hermes.deposit.delete_artifacts"]
+invenio = "hermes.commands.deposit.invenio:delete_artifacts"
+
+[tool.poetry.plugins."hermes.deposit.upload_artifacts"]
+invenio = "hermes.commands.deposit.invenio:upload_artifacts"
+
+[tool.poetry.plugins."hermes.deposit.publish"]
+invenio = "hermes.commands.deposit.invenio:publish"
```
Comment (led02) [inline pyproject.toml:111]: **chore:** We should put this an ADR

**Review by led02 — APPROVED** [review]
Some remarks may be followed up, but LGTM

File: .mailmap [hunk 4]
```diff
@@ -13,3 +13,5 @@
 # Real Name <email@address> Name, Real
 Jeffrey Kelling <j.kelling@hzdr.de> jkelling
 David Pape <d.pape@hzdr.de> David
+Stephan Druskat <stephan.druskat@dlr.de> Druskat, Stephan
+Michael Meinel <michael.meinel@dlr.de> Meinel, Michael
```
[the remark this answers is not in the document]
Comment (zyzzyxdonta) [reply .mailmap:17]: I'll move it into another PR 😁

[the remark this answers is not in the document]
Comment (jkelling) [reply .mailmap]: Merged #191 .

File: pyproject.toml (hunk 3 above)
Comment (zyzzyxdonta) [reply pyproject.toml:111]: → #192

**Pull request merged**